test/GogeDao.t.sol

```diff
@@ -319,19 +407,682 @@ contract DaoTest is Utility, Test {
+        assertEq(gogeToken.balanceOf(address(gogeDao)), joe_votes + gogeDao.minAuthorBal());
+
+        // Verify state. Poll should not be passed, though quorum has been met.
+        assertEq(gogeDao.polls(1, address(joe)), joe_votes);
+        assertEq(gogeDao.totalVotes(1), joe_votes + gogeDao.minAuthorBal());
+        assertEq(gogeDao.passed(1), false);
+        assertEq(gogeDao.pollEndTime(1), block.timestamp + 2 days);
+
+        // Verify quorum math.
+        uint256 num = gogeDao.getProportion(1);
+        assertTrue(num >= gogeDao.quorum()); // quorum met
+
+        // Verify poll has not been executed this Joe is not blacklisted, yet.
+        assertEq(gogeToken.isBlacklisted(address(joe)), false);
+
+        // Create gate keeper
+        gogeDao.updateGateKeeper(address(dev), true);
+
+        // Transfer tokens to gatekeeper
+        gogeToken.transfer(address(dev), 1 ether);
+        assertEq(gogeToken.balanceOf(address(dev)), 1 ether);
+
+        // gatekeeper adds vote to poll -> should pass poll
+        assert(dev.try_approveToken(address(gogeToken), address(gogeDao), 1 ether));
+        assert(dev.try_addVote(address(gogeDao), 1, 1 ether));
+
+        // Verify state change post poll being passed.
+        assertEq(gogeToken.balanceOf(address(dev)), 1 ether);
+        assertEq(gogeToken.balanceOf(address(joe)), joe_votes);
+        assertEq(gogeToken.balanceOf(address(gogeDao)), 0);
+        assertEq(gogeDao.polls(1, address(joe)), joe_votes);
+        assertEq(gogeDao.polls(1, address(this)), gogeDao.minAuthorBal());
+        assertEq(gogeDao.polls(1, address(dev)), 1 ether);
+        assertEq(gogeDao.totalVotes(1), joe_votes + 1 ether + gogeDao.minAuthorBal());
+        assertEq(gogeDao.passed(1), true);
+        assertEq(gogeDao.pollEndTime(1), block.timestamp);
+
+        // Verify poll has not been executed this Joe is not blacklisted, yet.
+        assertEq(gogeToken.isBlacklisted(address(joe)), true);
+    }
+
+    /// @notice Verify when payTeam is called the addresses held in teamMembers are paid out
+    function test_gogeDao_payTeam() public {
+        emit log_named_uint("BNB Balance", address(this).balance); // 79228162214.264337593543950335
+
+        // update governanceTokenAddr to address(this)
+        test_gogeDao_updateGovernanceToken();
+
+        // add team members
+        gogeDao.setTeamMember(address(jon), true);
+        gogeDao.setTeamMember(address(tim), true);
+
+        // update team balance on dao
+        payable(address(gogeDao)).transfer(1 ether);
+        gogeDao.updateTeamBalance(1 ether);
+
+        // Pre-state check
+        assertEq(address(gogeDao).balance, 1 ether);
+        assertEq(address(gogeDao).balance, gogeDao.teamBalance());
+        assertEq(address(jon).balance, 0 ether);
+        assertEq(address(tim).balance, 0 ether);
+
+        // pay team
+        gogeDao.payTeam();
+
+        // Post-state check
+        assertEq(address(gogeDao).balance, 0 ether);
```

Comment:
How would this equate to zero after the team is paid out? Balance should be equal to `preBalance % number of team members` which is the remainder of the division. This will also fail once the `teamBalance` issue is sorted in the actual contract lol.

Reply:
Incorrect. I'll show you why when we meet up